ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Util.hs

```diff
@@ -80,6 +81,21 @@ throwUnexpectedError :: ErrorHandling ImmutableDBError m
                      -> UnexpectedError -> m a
 throwUnexpectedError = throwError . handleUnexpected
 
+-- | Execute an action and catch the 'ImmutableDBError' and 'FsError' that can
+-- be thrown by it, and wrap the 'FsError' in an 'ImmutableDBError' using the
+-- 'FileSystemError' constructor.
+tryImmDB :: Monad m
```

Comment:
:+1:  Though perhaps also add "This should be used whenever ..." so people coming to this later know when to use it in new things too.